listener/src/event_handler/flux_transition_handler.rs

```diff
@@ -105,22 +90,18 @@ impl EventHandler for FluxTransitionHandler {
 #[cfg(test)]
 pub mod test {
     use super::*;
-    use graph_mock::MockStore;
+    use dfusion_core::database::tests::DbInterfaceMock;
+    use dfusion_core::database::*;
+    use dfusion_core::models::{PendingFlux, AccountState};
     use web3::types::{Bytes, H256};
 
     #[test]
     fn test_applies_deposits_existing_state() {
-        let schema = util::test::fake_schema();
-        let store = Arc::new(MockStore::new(vec![(schema.id.clone(), schema)]));
-        let handler = FluxTransitionHandler::new(store.clone());
+        let store = Arc::new(DbInterfaceMock::new());
 
         // Add previous account state and pending deposits into Store
         let existing_state = AccountState::new(H256::zero(), U256::zero(), vec![0, 0, 0, 0], 1);
-        let entity = existing_state.into();
-        store.apply_entity_operations(vec![EntityOperation::Set {
-            key: util::entity_key("AccountState", &entity),
-            data: entity
-        }], None).unwrap();
+        store.get_balances_for_state_index.given(U256::zero()).will_return(Ok(existing_state));
```

Comment:
nit: I'm usually more in favour in not chaining lot's of things in the same line. Even for tests. Otherwise you need to read all of it from left to right to understand.

I think in the book they also recommend to do only one thing in every line
```
store.get_balances_for_state_index
  .given(U256::zero())
  .will_return(Ok(existing_state)); 
```